contracts/oracle/FastPriceFeed.sol

```diff
@@ -13,18 +14,34 @@ pragma solidity 0.6.12;
 contract FastPriceFeed is ISecondaryPriceFeed, IFastPriceFeed, Governable {
     using SafeMath for uint256;
 
+    // fit data in a uint256 slot to save gas costs
+    struct PriceDataItem {
+        uint160 refPrice; // Chainlink price
+        uint32 refTime; // last updated at time
+        uint32 cumulativeRefDelta; // cumulative Chainlink price delta
+        uint32 cumulativeFastDelta; // cumulative fast price delta
+    }
+
     uint256 public constant PRICE_PRECISION = 10 ** 30;
 
+    uint256 public constant CUMULATIVE_DELTA_PRECISION = 10 * 1000 * 1000;
+
+    uint256 public constant MAX_REF_PRICE = type(uint160).max;
+    uint256 public constant MAX_CUMULATIVE_REF_DELTA = type(uint32).max;
+    uint256 public constant MAX_CUMULATIVE_FAST_DELTA = type(uint32).max;
+
     // uint256(~0) is 256 bits of 1s
     // shift the 1s by (256 - 32) to get (256 - 32) 0s followed by 32 1s
-    uint256 constant public PRICE_BITMASK = uint256(~0) >> (256 - 32);
+    uint256 constant public BITMASK_32 = uint256(~0) >> (256 - 32);
```

Comment:
i know it's an old code, but probably better to use just `2 ** 32 - 1` instead? each time i see that code i stop to parse it :)